src/lib/core/CHIPError.h

```diff
@@ -123,23 +131,39 @@ class ChipError
      * @note
      *  The result is valid only if CanEncapsulate() is true.
      */
+#if __cplusplus >= 202002L
+    constexpr ChipError(Range range, ValueType value) :
+        mError(MakeInteger(range, (value & MakeMask(0, kValueLength)))) CHIP_INITIALIZE_ERROR_SOURCE(nullptr, 0, std::source_location())
+    {}
+    constexpr ChipError(Range range, ValueType value, const char * file, unsigned int line, std::source_location location = std::source_location::current()) :
```

Comment:
So how does this handle situations where the ChipError() constructor is called via a macro defined at location A but used at location B?  Which location ends up in `location`?

Reply:
It should always show location B: https://godbolt.org/z/oThxMKKM4

Reply:
Excellent, thanks!